ocaml/xapi/xapi_pci_helpers.ml

```diff
@@ -172,3 +173,62 @@ let get_host_pcis () =
 let igd_is_whitelisted ~__context pci =
   let vendor_id = Db.PCI.get_vendor_id ~__context ~self:pci in
   List.mem vendor_id !Xapi_globs.igd_passthru_vendor_whitelist
+
+let is_pci_hidden_cmdline ~__context ~self =
+  let cmdline =
+    match Unixext.read_lines ~path:"/proc/cmdline" with
+    | [x] ->
+        x
+    | _ ->
+        failwith "Unable to read cmdline"
+  in
+  let device = Db.PCI.get_pci_id ~__context ~self in
+  let elems = String.split_on_char ' ' cmdline in
+
+  let pciback =
+    List.find_opt
+      (fun s -> String.starts_with ~prefix:"xen-pciback.hide=" s)
+      elems
```

Comment:
It's worth using `List.find_map` to remove `"xen-pciback.hide="` from the result.

I recommend defining `xen_hide_param` and `xen_hide_param_length` at the top level and using taking out the substring with after the prefix check:
`String.sub s xen_hide_param_length ((String.length s) - xen_hide_param_length)`